crates/runtime/src/dataaccelerator/sqlite.rs

```diff
@@ -88,18 +105,68 @@ impl DataAccelerator for SqliteAccelerator {
         "sqlite"
     }
 
+    /// Initializes an SQLite database for the dataset
+    /// If the dataset is not file-accelerated, this is a no-op
+    /// This step is required for federation, as SQLite connections attach to all other configured SQLite databases.
+    /// Federation then requires that all attached databases exist before dataset registration.
+    async fn init(
+        &self,
+        dataset: &Dataset,
+    ) -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
+        let path = self.sqlite_file_path(dataset);
+
+        if let (Some(path), Some(acceleration)) = (&path, &dataset.acceleration) {
+            if !acceleration.params.contains_key("sqlite_file") {
+                make_spice_data_directory()
+                    .map_err(|err| Error::AccelerationCreationFailed { source: err.into() })?;
+            }
+
+            SqliteConnectionPool::init(path, acceleration.mode.to_string().as_str().into())
+                .await
+                .context(AccelerationInitializationFailedSnafu)?;
+        }
+
+        Ok(())
+    }
+
     /// Creates a new table in the accelerator engine, returning a `TableProvider` that supports reading and writing.
     async fn create_external_table(
         &self,
         cmd: &CreateExternalTable,
+        dataset: Option<&Dataset>,
     ) -> Result<Arc<dyn TableProvider>, Box<dyn std::error::Error + Send + Sync>> {
+        let mut cmd = cmd.clone();
         if !cmd.options.contains_key("file") {
             make_spice_data_directory()
                 .map_err(|err| Error::AccelerationCreationFailed { source: err.into() })?;
         }
 
+        if let Some(Some(attach_databases)) = dataset.map(|this_dataset: &Dataset| {
+            this_dataset.app.as_ref().map(|a| {
+                let datasets = Runtime::get_valid_datasets(a, crate::LogErrors(false));
+                datasets
+                    .iter()
+                    .filter(|d| {
+                        d.acceleration
+                            .as_ref()
+                            .map_or(false, |a| a.engine == Engine::Sqlite)
+                    })
+                    .filter_map(|other_dataset| {
+                        if **other_dataset == *this_dataset {
+                            None
+                        } else {
+                            self.sqlite_file_path(other_dataset)
+                        }
+                    })
+                    .collect::<Vec<String>>()
+            })
+        }) {
+            cmd.options
+                .insert("attach_databases".to_string(), attach_databases.join(";"));
+        }
```

Comment:
It feels weird to have this Sqlite specific code in the generic DataAccelerator path. Would a middleware style approach work better? (i.e. when registering a DataAccelerator, it can opt to register a middleware handle that runs here and has the option to modify `cmd`)